### PR TITLE
Improve proxy server usage

### PR DESCRIPTION
**Pull request by AllentDan**

### CODE REVIEW

**Comment by lvhan028** [comment]
Let's add another CLI for `python3 -m lmdeploy.serve.proxy.proxy --server_name {server_name} --server_port {server_port} --strategy "min_expected_latency"`
```shell
lmdeploy serve proxy --server-name {server_name} --server-port {server_port} --strategy "min_expected_latency"
```

File: docs/en/llm/proxy_server.md [hunk 1]
```diff
@@ -30,6 +32,65 @@ APIs related to usage include:
 
 The usage of these APIs is the same as that of api_server.
 
+### add, delete and query through commands
```
Comment (lvhan028) [inline docs/en/llm/proxy_server.md:35]: commands -> `curl`

File: docs/en/llm/proxy_server.md [hunk 2]
```diff
@@ -30,6 +32,65 @@ APIs related to usage include:
...
+```
+
+```shell
+curl -X 'POST' \
+  'http://localhost:10086/nodes/remove?node_url=http://0.0.0.0:23333' \
+  -H 'accept: application/json' \
+  -d ''
+```
+
+### add, delete and query through python
```
Comment (lvhan028) [inline docs/en/llm/proxy_server.md:60]: ### Node Management through python

File: docs/en/llm/proxy_server.md [hunk 3]
```diff
@@ -30,6 +32,65 @@ APIs related to usage include:
...
+  -H 'accept: application/json' \
+  -H 'Content-Type: application/json' \
+  -d '{
+  "url": "http://0.0.0.0:23333"
+}'
+```
+
+```shell
+curl -X 'POST' \
+  'http://localhost:10086/nodes/remove?node_url=http://0.0.0.0:23333' \
```
Comment (lvhan028) [inline docs/en/llm/proxy_server.md:55]: Which is a more conventional way, "remove/?node_url" or set the parameter in `-d` field?

Comment (AllentDan) [reply docs/en/llm/proxy_server.md]: Both worked, the command was from swagger UI.

File: lmdeploy/serve/openai/api_server.py [hunk 4]
```diff
@@ -40,6 +40,9 @@ class VariableInterface:
     session_id: int = 0
     api_keys: Optional[List[str]] = None
     request_hosts = []
+    # following are for registering to proxy server
+    proxy_url: Optional[str] = None
+    api_server_url: Optional[str] = None
```
Comment (lvhan028) [inline lmdeploy/serve/openai/api_server.py:45]: Is it necessary? Can we use `http://{server_ip}:{server_port}` instead

**Review by lvhan028 — APPROVED** [review]

File: lmdeploy/serve/openai/api_server.py (hunk 4 above)
Comment (AllentDan) [reply lmdeploy/serve/openai/api_server.py:45]: It is a global variable. `VariableInterface.api_server_url = f'{http_or_https}://{server_name}:{server_port}'`

File: lmdeploy/cli/serve.py [hunk 5]
```diff
@@ -204,6 +208,36 @@ def add_parser_api_client():
...
+                            type=int,
+                            default=8000,
+                            help='Server port of the proxy')
+        parser.add_argument(
+            '--strategy',
+            type=str,
+            choices=['random', 'min_expected_latency', 'min_observed_latency'],
+            default='min_expected_latency',
+            help='the strategy to dispatch requests to nodes')
+        parser.add_argument('--api-key',
```
Comment (RunningLeon) [inline lmdeploy/cli/serve.py:234]: should it be `--api-keys` and could use `ArgumentHelper.api_keys(parser)`?
https://github.com/InternLM/lmdeploy/blob/8173929e45bda0a8074be9af631a5f30fc8185ea/lmdeploy/serve/proxy/proxy.py#L482

**Review by RunningLeon — APPROVED** [review]
LGTM

**Pull request merged**